### PR TITLE
Defrates1/more updates

### PR DESCRIPTION
**Pull request by cdefrates**
- TOSS-04-020230 - set disabled users that do not have home directories to "non-interactive"
- TOSS-04-020300 - update to allow for "home directories must have mode 0770 or less permissive."
- TOSS-04-020320 and 020210 - break up the 'awk' shell command to avoid hitting file descriptor limit on large nodes
- TOSS-04-040630 and 040640 - set option to create missing file in order to fix failing control

### CODE REVIEW

**Comment by IanLee1521** [comment]
@cdefrates - fyi, it looks like this has conflicts. Probably from me just having merged #41

File: ansible/roles/stig/tasks/TOSS_04_020230.yml [hunk 1]
```diff
@@ -14,23 +14,29 @@
...
+    - name: Create local_users variable from the getent output
+      ansible.builtin.set_fact:
+        local_users: '{{ ansible_facts.getent_passwd | dict2items }}'
+    - name: Gather the home directories of local, interactive users
+      ansible.builtin.stat:
+        path: '{{ item.value.4 }}'
+      register: home_dirs
+      loop: '{{ local_users }}'
+      when:
+        - item.value.1 | int >= 1000 and '/nologin' not in item.value.5 and '/false' not in item.value.5
```
Comment (IanLee1521) [inline ansible/roles/stig/tasks/TOSS_04_020230.yml:30]: Might not be an issue, but I've also seen `/dev/null` used before to mean "disabled"

File: ansible/roles/stig/tasks/TOSS_04_020210.yml [hunk 2]
```diff
@@ -15,20 +15,19 @@
...
       changed_when: false
       failed_when: home_dirs.rc not in [0,1]
       check_mode: false
-    - name: Change the owner of the local interactive user's home directory
+    - name: Set the group of the local interactive user's home directory
       ansible.builtin.file:
         path: '{{ item.split().1 }}'
         group: '{{ item.split().2 }}'
       loop: '{{ home_dirs.stdout_lines }}'
+      failed_when: false
```
Comment (IanLee1521) [inline ansible/roles/stig/tasks/TOSS_04_020210.yml:30]: I would have thought this would be the default behavior, no?

Comment (IanLee1521) [reply ansible/roles/stig/tasks/TOSS_04_020210.yml:30]: Got it, thanks!

**Comment by cdefrates** [comment]
Added a few more small changes after rerunning the entire playbook. Changes included updating a variable name, adding a check_mode option, adding a new condition to check for interactive users, and made some regexps more specific to avoid duplicating a line instead of overwriting.

**Pull request merged**